android/Idviu/app/src/main/java/com/idviu/sample/hss/HSSPlayerActivity.java

```diff
@@ -0,0 +1,288 @@
+		mPlayerView.setWidevineOfflineMode(false); //set to true if you want to retrieve a persistent license for Widevine
+		mPlayerView.setOnPreparedListener(new OnPreparedListener() {
+
+			@Override
+			public void onPrepared(MediaPlayer mp) {
+				Log.i(TAG, "redirected url: " + mPlayerView.getPlayer().getRedirectedUrl());
+			}
+		});
+		HSSCustomLicenseRequestHandler.instance().setPlayerView(mPlayerView);
+
+		Intent intent = this.getIntent();
+		Bundle bundle = intent.getExtras();
+		Entry entry = (Entry)bundle.getSerializable("entry");
+		if (mPlayerView.getPlaylist().getSize() == 0 && entry != null ){ //only if empty
+			HSSPlaylistItem item = null;
+				if (entry.downloadId >= 0){
+					item = HSSPlaylistItem.createWithDownloadId(entry.downloadId);
+					if (entry.downloadForceLocalMode)
+						item.setForceLocalMode(entry.downloadForceLocalMode);
+					mPlayerView.getPlaylist().addItem(item);
+				}else {
+					if (entry.streamrootDnaEnabled) {
+						startDnaClient(entry);
+						item = HSSPlaylistItem.createWithUrl(dnaClient.getManifestUrl().toString());
+					} else {
+						item = HSSPlaylistItem.createWithUrl(entry.getUrl());
+					}
+					if(entry.getStartTime() != 0)
+						item.setStartPosition(entry.getStartTime());
+					if(entry.getBonusTime() != 0)
+						item.setBonusDRMTime(entry.getBonusTime());
+
+					item.setPlayreadyLicenseUrlAndCustomData(entry.getPlayreadyUrl(), entry.getPlayreadyCustomData());
+					item.setWidevineLicenseUrl(entry.getWidevineUrl());
+					item.setWidevineCustomData(entry.getWidevineCustomData());
+					item.setMarlinLicenseToken(entry.getMarlinUrl());
+					if (entry.downloadForceLocalMode)
+						item.setForceLocalMode(entry.downloadForceLocalMode);
+					if (entry.getMaxVideoBitrate() != 0)
+						item.setMaxVideoBitrate(entry.getMaxVideoBitrate());
+					if (entry.getUserAgent() != null && entry.getUserAgent() != "" )
+						item.setUserAgent(entry.getUserAgent());
+					item.setPlayerParameters(entry.getPlayerParameters());
+
+					item.setFilteredHeaders(entry.getFilteredHeaders());
+					for (String keys : entry.getHeaders().keySet())
+						item.addLicenseHeader(keys, entry.getHeaders().get(keys));
+
+					mPlayerView.getPlaylist().addItem(item);
+				}
+		}
+	}
+
+
+	/**
+	 * Start Streamroot DNAClient
+	 */
+	private void startDnaClient(Entry entry) {
+		// Init the player interactor
+		PlayerInteractor playerInteractor = new IdviuInteractor(this.mPlayerView);
+
+		// Init the player bandwidth listener
+		BandwidthListener bandwidthListener = new IdviuBandwithMeter(this.mPlayerView);
+
+		// Init qosModule (optional)
+		QosModule qosModule = new IdviuQoSModule(this.mPlayerView);
+
+		// Build and start the DNAClient
+		try {
+			this.dnaClient = DnaClient.newBuilder()
+					.context(getApplicationContext())
+					.playerInteractor(playerInteractor)
+					.bandwidthListener(bandwidthListener)
+					.qosModule(qosModule)
+					.latency(20)
+					.contentId(entry.getTitle())
```

Comment:
Why not for the sake of the example but can be dangerous if poorly used and misunderstood by the final integrator.